docs/website/contents/for-developers/CivicTime.md

```diff
@@ -0,0 +1,193 @@
+    So the lower bound on translations was at least 1.5 days ahead, since the post-Byron stability window has always been 3k/f slots.
+
+  - Since Conway, it's been one epoch = 10k/f slots = 432000 slots = 432000 seconds = 5 days.
+    So the lower bound on translations is currently at least 3.5 days ahead, since the stability window is still 3k/f slots.
+
+Those lower bounds are for an arbitrary ledger state, and so they're the "worst-case".
+For specific ledger states, their detailed location in the epoch can increase the lower bound by up to one epoch, which has always been 5 days in every era so far.
+
+The limit is one stability window less the the post-voting buffer because --- assuming only the Praos security argument --- that's the upper bound on the age of a block the honest Praos node might need to discard as part of a rollback.
+In particular, until the oldest such block is after the voting deadline, a rollback could switch to a chain with a different voting outcome in that epoch.
+For example, switching from a chain in which the next era is only a few slots away to one in which it is at least another epoch away.
+
+*Aside*.
+On the other hand, when using a (non-recent) ledger state in a historical era, the HFC could theoretically use its knowledge about the upcoming eras in order to translate even further ahead.
+For example, the HFC could safely assume the one second duration for slots between some given Babbage ledger state and however many slots it might possibly take before that ledger state could transition from Babbage to Conway and from Conway to whatever comes after, because that mystery era is the first time the slot duration could change.
+The HFC does not do this in practice, since the extra complexity is not worthwhile; it's not necessary for syncing nodes to be smarter than caught-up nodes.
+
+[^EmergencyRapidity]: Technicality.
+    It could perhaps be weeks or days in an emergency, but a change to the slot duration would almost certainly be avoided in that case.
+
+## What About Chain Growth Failures?
+
+Roughly, Today's HFC was derived as follows.
+
+  - Because of humans' general assumptions about time, it would be prohibitively confusing if the node (even its internal interfaces) might give different translations of the same slots/civic times.
+    At the very least, the node should refuse to do a translation that might be invalidated by on-chain governance in the meantime (eg several weeks into the future).
+
+  - However, the node should be able to translate slot/civic times from the near future.
+    At the very least, some users presumably want some capacity to plan ahead (eg to smartly schedule their node's brief downtime).
+
+  - The node should even refuse to do translations that might be invalidated by a rollback.
+
+  - Even a violation of Praos's Chain Growth property would not excuse violating the above constraints.
```

Comment:
> Even a violation of Praos's Chain Growth property would not excuse violating the above constraints.

Are we sure about this? I assume that you list this to explain block counting; but I think the most likely reason for block counting is that switching to a denser chain can't result in being able to translate fewer slots (ie Section 17.4.3 "Stability should not be approximated" in the report), and the special behavior around chain growth violations is just non-interntional/accidental/emergent. Ie that section says (emphasis mine):

> The consensus layer **therefore** does not use the ledger’s layer notion of stability, but instead maintains additional state so that it can use block numbers rather than slot numbers to determine stability and be able to determine stability precisely rather than approximate it.

Reply:
Thanks for pointing me back to The Report.

I rewrote the latter parts of this section to better match that.

In the process, I (re-?)realized that today's HFC will give disagreeing translations due to switching from a chain that violates Chain Growth (where the HFC overrides an era transition) to one that does not (where the HFC does not override the era transition). Which ends up making the content in this section feel much less linear/monotone, sadly.

Reply:
Oh yeah, good point about block counting not even accomplishing the "time translation is stable even during chain growth violations" goal :+1: